input/kafkamdm/kafkamdm.go

```diff
@@ -225,14 +225,17 @@ func (k *KafkaMdm) consumePartition(topic string, partition int32, currentOffset
 	partitionLogSizeMetric := partitionLogSize[partition]
 	partitionLagMetric := partitionLag[partition]
 
-	partitionOffsetMetric.Set(int(currentOffset))
-	// we need the currentLogSize to be able to record our inital Lag.
-	offset, err := k.client.GetOffset(topic, partition, sarama.OffsetNewest)
-	if err != nil {
-		log.Error(3, "kafka-mdm failed to get log-size of partition %s:%d. %s", topic, partition, err)
-	} else {
-		partitionLogSizeMetric.Set(int(offset))
-		partitionLagMetric.Set(int(offset - currentOffset))
+	if currentOffset >= 0 {
```

Comment:
from the below comment.  "we cant set the offsetMetrics until we know what offset we are at."

the stats package uses unsigned INTs. So setting a value to a negative results in a really big number being assigned, which causes havoc in graphs.